tests/components/vacuum/__init__.py

```diff
@@ -1 +1,124 @@
 """The tests for vacuum platforms."""
+
+from typing import Any
+
+from homeassistant.components.vacuum import (
+    DOMAIN,
+    STATE_CLEANING,
+    STATE_DOCKED,
+    STATE_IDLE,
+    STATE_PAUSED,
+    STATE_RETURNING,
+    StateVacuumEntity,
+    VacuumEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from tests.common import (
+    MockConfigEntry,
+    MockEntity,
+    MockModule,
+    MockPlatform,
+    mock_integration,
+    mock_platform,
+)
+
+
+class MockVacuum(MockEntity, StateVacuumEntity):
+    """Mock vacuum class."""
+
+    _attr_supported_features = (
+        VacuumEntityFeature.PAUSE
+        | VacuumEntityFeature.STOP
+        | VacuumEntityFeature.RETURN_HOME
+        | VacuumEntityFeature.FAN_SPEED
+        | VacuumEntityFeature.BATTERY
+        | VacuumEntityFeature.CLEAN_SPOT
+        | VacuumEntityFeature.MAP
+        | VacuumEntityFeature.STATE
+        | VacuumEntityFeature.START
+    )
+    _attr_battery_level = 99
+    _attr_fan_speed_list = ["slow", "fast"]
+
+    def __init__(self, **values: Any) -> None:
+        """Initialize a mock vacuum entity."""
+        super().__init__(**values)
+        self._attr_state = STATE_DOCKED
+        self._attr_fan_speed = "slow"
+
+    def stop(self, **kwargs: Any) -> None:
+        """Stop cleaning."""
+        self._attr_state = STATE_IDLE
+
+    def return_to_base(self, **kwargs: Any) -> None:
+        """Return to base."""
+        self._attr_state = STATE_RETURNING
+
+    def clean_spot(self, **kwargs: Any) -> None:
+        """Clean a spot."""
+        self._attr_state = STATE_CLEANING
+
+    def set_fan_speed(self, fan_speed: str, **kwargs: Any) -> None:
+        """Set the fan speed."""
+        self._attr_fan_speed = fan_speed
+
+    def start(self) -> None:
+        """Start cleaning."""
+        self._attr_state = STATE_CLEANING
+
+    def pause(self) -> None:
+        """Pause cleaning."""
+        self._attr_state = STATE_PAUSED
+
+
+async def create_entity(
```

Comment:
Can we use `setup_test_component_platform` instead?  The test can create the entity directly as needed by calling `MockVacuum`.

Reply:
I can try to make that change tomorrow!

Reply:
Do you want create_entity to use setup_test_component_platform inside of it? Or replace create entity with setup_test_componeont_platform and calling MockVacuum?

Reply:
I'd remove create entity.

Reply:
Done!